packages/taquito/src/operations/types.ts

```diff
@@ -563,6 +566,53 @@ export interface FailingNoopParams {
   basedOnBlock: BlockIdentifier;
 }
 
+export interface InternalStakingParams {
+  source: string;
+  amount: number;
+  fee?: number;
+  gasLimit?: number;
+  storageLimit?: number;
+  mutez?: boolean;
+  parameter: TransactionOperationParameter;
+}
+
+type PartialBy<T, K extends keyof T> = Omit<T, K> & Partial<Pick<T, K>>;
```

Comment:
can we add a comment explaining what PartialBy does?

Reply:
If someone knows `Partial` helper type in typescript, the name should be self explanatory. I added a comment anyway.